CMakeLists.txt

```diff
@@ -28,42 +28,56 @@ cmake_minimum_required(VERSION 3.2)
 
 project(VulkanHppGenerator)
 
-file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Docs/src/spec/vk.xml vk_spec)
-string(REPLACE "\\" "\\\\" vk_spec ${vk_spec})
-add_definitions(-DVK_SPEC="${vk_spec}")
+# If you wish to target a specific version of the spec, override this value like so 
+# cmake -DVK_SPEC_URL:STRING=https://example.com/some/other/url/vk.xml
+set(VK_SPEC_URL "https://raw.githubusercontent.com/KhronosGroup/Vulkan-Docs/1.0/src/spec/vk.xml" CACHE STRING "URL from which to fetch the Vulkan Spec XML")
 
-file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/vulkan/vulkan.hpp vulkan_hpp)
-string(REPLACE "\\" "\\\\" vulkan_hpp ${vulkan_hpp})
-add_definitions(-DVULKAN_HPP="${vulkan_hpp}")
-
-set(HEADERS
-)
+# Download the request URL specification
+file(DOWNLOAD ${VK_SPEC_URL} ${CMAKE_CURRENT_BINARY_DIR}/vk.xml )

```

Comment:
This way of fetching vk.xml does not update the spec automatically if you change the CMakeLists.txt. To update one would have to delete the cache and configure the project again.

I'm wondering if it'd make sense to keep the vk.xml as external subproject and let an application download a different spec if required. The build process would be:

Download Vulkan-Hpp project (as zip)
Download your favorite vk.xml to some location
unzip Vulkan-Hpp
configure & vbuild vulkan.hpp
run VulkanHppGenerator <path to your favorite vk.xml>

As alternate solution the url of the spec could be in a separate file and a custom target with dependency on that file would download vk.xml during the build process. VulkanHppGenerator would have to depend on that custom target.